flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -614,7 +563,9 @@ private Collection<Integer> legacyTransform(Transformation<?> transform) {
         if (transform.getBufferTimeout() >= 0) {
             streamGraph.setBufferTimeout(transform.getId(), transform.getBufferTimeout());
         } else {
-            streamGraph.setBufferTimeout(transform.getId(), defaultBufferTimeout);
+            streamGraph.setBufferTimeout(
+                    transform.getId(),
+                    configuration.get(ExecutionOptions.BUFFER_TIMEOUT).toMillis());
```

Comment:
Should we consider `BUFFER_TIMEOUT_ENABLED` here?

Reply:
Good catch, there was an issue with my handling of BUFFER_TIMEOUT_ENABLED. In fact, what I intended to implement was that whenever getting the buffer timeout from the configuration, we need to first check BUFFER_TIMEOUT_ENABLED; the same applies to setting the buffer timeout in the configuration. Thanks for pointing out the issue, it has now been fixed.